src/PIL/Image.py

```diff
@@ -3186,6 +3204,25 @@ def __array_interface__(self) -> dict[str, Any]:
         raise NotImplementedError()
 
 
+class SupportsArrowArrayInterface(Protocol):
+    """
+    An object that has an ``__arrow_c_array__`` method corresponding to the arrow c
+    data interface.
+    """
+
+    # Sorry, no types for you until pre-commit.ci stops changing stringy
+    # PyCapsule type to an unimportable value type, which then fails lint.
+    # PyCapsules are not importable, and only available in the C-API layer.
+    # def __arrow_c_array__(
+    #     self, requested_schema: 'PyCapsule' = None
+    # ) -> tuple['PyCapsule', 'PyCapsule']:
+    #     raise NotImplementedError()
+
+    # old not typed definition.
+    def __arrow_c_array__(self, requested_schema=None):
+        raise NotImplementedError()
```

Comment:
Here's one way:

```suggestion
    def __arrow_c_array__(
        self, requested_schema: "PyCapsule" = None  # type: ignore[name-defined]  # noqa: F821, UP037
    ) -> tuple["PyCapsule", "PyCapsule"]:  # type: ignore[name-defined]  # noqa: F821, UP037
        raise NotImplementedError()
```
Perhaps we could use a pyi file, but let's not block on this.